### PR TITLE
Preview build

### PR DESCRIPTION
**Pull request by colearendt**
create a copy of the "latest" workflow to build preview/daily images

add a "get-version.py" script to help retrieve versions in an easier-to-test / easier-to-understand fashion

add the Connect content init build for preview images

### CODE REVIEW

**Review by aronatkins — APPROVED** [review]

File: .github/workflows/build-preview.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      matrix:
+        config:
+          - {product: 'rstudio-workbench-preview', dir: 'server-pro', prefix: RSP, os: 'bionic/amd64', version: 'daily'}
+          - {product: 'rstudio-workbench-preview', dir: 'server-pro', prefix: RSP, os: 'bionic/amd64', version: 'preview'}
+          - {product: 'rstudio-connect-preview', dir: 'connect', prefix: RSC, version: 'daily'}
+          - {product: 'rstudio-connect-content-init-preview', dir: 'connect-content-init', prefix: RSC, version: 'daily'}
+          - {product: 'r-session-complete-preview', tag_prefix: 'bionic-', dir: 'r-session-complete/bionic', os: 'bionic/amd64', prefix: RSP, version: 'daily'}
+          - {product: 'r-session-complete-preview', tag_prefix: 'centos7-', dir: 'r-session-complete/centos7', os: 'centos7/x86_64', prefix: RSP, version: 'daily'}
+          - {product: 'r-session-complete-preview', tag_prefix: 'bionic-', dir: 'r-session-complete/bionic', os: 'bionic/amd64', prefix: RSP, version: 'preview'}
+          - {product: 'r-session-complete-preview', tag_prefix: 'centos7-', dir: 'r-session-complete/centos7', os: 'centos7/x86_64', prefix: RSP, version: 'preview'}
```
Comment (bdeitte) [inline .github/workflows/build-preview.yaml:27]: Can we (or are you planning to) document somewhere these new builds?

File: .github/workflows/build-preview.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Login to ghcr.io
+        if: ${{ github.ref == 'refs/heads/main' }}
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.BUILD_PAT }}
+
```
Comment (bdeitte) [inline .github/workflows/build-preview.yaml:106]: All of the above feels like a lot of duplication from the other build file.  Don't have any solution I know from GitHub Actions but wanted to complain anyways.  😄

**Comment by bdeitte** [comment]
A late review but looks good!

File: .github/workflows/build-preview.yaml (hunk 2 above)
Comment (colearendt) [reply .github/workflows/build-preview.yaml:106]: YES! Same haha. I actually vim-diff'd to make sure they were nearly identical. The "right" pattern is probably to separate these into their own actions, but I'm not much of a typescript developer 😛

File: .github/workflows/build-preview.yaml (hunk 1 above)
Comment (colearendt) [reply .github/workflows/build-preview.yaml:27]: Yeah. Docs are next 😄  Most of these builds are not new, FWIW. They are just used almost exclusively internally for QA testing of preview releases, etc.

**Pull request merged**